prboom2/src/gl_main.c

```diff
@@ -2193,10 +2193,11 @@ static void gld_AddFlat(int sectornum, dboolean ceiling, visplane_t *plane)
 
   indexed = V_IsWorldLightmodeIndexed();
 
+  if (plane->picnum & PL_SKYFLAT || plane->picnum == skyflatnum) // don't draw if sky
```

Comment:
Are we sure the PL_SKYFLAT bits are always there?

Reply:
Yeah, this is the same logic used for detecting sky planes elsewhere, e.g. in `R_FindPlane`.

Reply:
There is a texture flag for skies as well, and that flag is not always set for sky textures. I learned to never trust anything in this code base, no matter how obvious 😄

Reply:
If you look at the logic in `R_Subsector`, it passes `frontsector->sky` to `R_FindPlane` if and only if it has `PL_SKYFLAT` set and the floor/ceiling is `skyflatnum`, and just `floorpic`/`ceilingpic` otherwise.  `R_FindPlane` sets this in the resulting `visplane_t` verbatim.  So `PL_SKYFLAT` should always be set for sectors with non-standard skies.